crates/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -269,6 +302,38 @@ impl<'a> FunctionContext<'a> {
         Self::map_type(element_type, |typ| {
             let offset = self.make_offset(base_index, field_index);
             field_index += 1;
+
+            let array_type = &self.builder.type_of_value(array);
+            match array_type {
+                // Prepare a slice access
+                // Check that the index being used is less than the dynamic slice length
+                Type::Slice(_) => {
+                    let array_len =
+                        max_length.expect("ICE: a length must be supplied for indexing slices");
```

Comment:
Can we break this chunk into another function? Everything inside the `Type::Slice` match I suppose?

Reply:
will do